api/v1/runtimecomponent_types.go

```diff
@@ -765,6 +765,17 @@ func (cr *RuntimeComponent) GetManageTLS() *bool {
 	return cr.Spec.ManageTLS
 }
 
+func (cr *RuntimeComponent) GetManagedPort() int {
+	return int(cr.GetService().GetPort())
```

Comment:
Validate that the default value Operator sets (when user hasn't configured Service port) is retrieved here.

Reply:
Added nil check on `GetService()` and fallback to `8080`